app/controllers/api/v1/batches_controller.rb

```diff
@@ -0,0 +1,241 @@
+      # | `project_id` | The id of a Project.                                                                    |
+      # | `api_token`  | The api token for a Project. Not required if the request is coming from internal views. |
+      # | `name`       | The `name` of the Batch.                                                              |
+      #
+      # Body Parameters
+      # ---------------
+      #
+      # |                   |                                                                                                               |
+      # |:------------------|:----------------------------------------|
+      # | `article_names`   | The list of article names in the batch  |
+
+      def update
+        batch_name = params['name']
+        ArticleBatch.transaction do
+          # delete articles from current batch
+          ArticleBatch.where(batch_name: batch_name).delete_all
+
+          # add new articles to current batch.
+          errors = create_article_batch(batch_name, params['article_names'])
+          if errors
+            render_errors(errors)
+            return
+          end
+        end
+
+        respond_with @project do |format|
+          format.json { render json: decorate_batch(batch_name) }
+        end
+      end
+
+      def destroy
+        batch_name = params['name']
+        ArticleBatch.where(batch_name: batch_name).delete_all
+
+        respond_with @project do |format|
+          format.json { render json: nil }
+        end
+      end
+
+      # private
+
+      # ===== START AUTHENTICATION/AUTHORIZATION/VALIDATION ============================================================
+      def authenticate_with_api_token!
+        unless Project.where(id: params[:project_id], api_token: params[:api_token]).exists?
+          render_error_with_message(t('controllers.api.v1.batches.invalid_api_token'), :unauthorized)
+        end
+      end
+
+      def find_project
+        @project = Project.find_by_id(params[:project_id])
+
+        unless @project
+          render_error_with_message(t('controllers.api.v1.batches.project_not_found'), :not_found)
+        end
+      end
+
+      def find_batch
+        unless find_batch_by_name?(params[:name])
+          render_error_with_message(t('controllers.api.v1.batches.batch_not_found'), :not_found)
+        end
+      end
+
+      def api_request?
+        params[:api_token].present?
+      end
+
+      # ===== END AUTHENTICATION/AUTHORIZATION/VALIDATION ==============================================================
+
+      # ===== START DECORATORS =========================================================================================
+      def decorate_batches
+        ArticleBatch.where(article: @project.articles).map(&:batch_name).uniq
+      end
+
+      def decorate_batch(batch_name)
+        ArticleBatch.where(batch_name: batch_name).order(id: :asc).map(&:article).map(&:name)
```

Comment:
It would be really nice having also the `ready` status of each article in the API response.